nodejs/eks/cluster.ts

```diff
@@ -477,26 +477,31 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
             // endpoint, and verify that it is reachable.
             const healthz = `${clusterEndpoint}/healthz`;
             const agent = createHttpAgent(args.proxy);
-            for (let i = 0; i < 60; i++) {
+            const maxRetries = 60;
+            const timeoutMilliseconds = 5000;
+            for (let i = 0; i < maxRetries; i++) {
                 try {
                     await new Promise((resolve, reject) => {
                         const options = {
                             ...url.parse(healthz),
                             rejectUnauthorized: false, // EKS API server uses self-signed cert
                             agent: agent,
+                            timeout: 1000,
                         };
                         const req = https
                             .request(options, res => {
                                 res.statusCode === 200 ? resolve() : reject(); // Verify healthz returns 200
                             });
+                        req.on("timeout", reject);
                         req.on("error", reject);
                         req.end();
                     });
                     break;
                 } catch (e) {
-                    pulumi.log.info(`Waiting for cluster endpoint (${i + 1})`, eksCluster, undefined, true);
+                    const retrySecondsLeft = (maxRetries - i) * timeoutMilliseconds / 1000;
+                    pulumi.log.info(`Waiting for cluster endpoint (${i + 1}). Waiting up to (${retrySecondsLeft}) more seconds...`, eksCluster, undefined, true);
```

Comment:
I think the retry count is somewhat confusing in the current string. Should either remove it if the timeout is sufficient, or show progress towards the max (retry 2/60) or similar.